### ISSUE TITLE
Division by zero

### ISSUE THREAD

**Issue by haesleinhuepf** [clEsperanto/pyclesperanto_prototype#3]
Dear cle friends, dear future-self,

Division by zero on the GPU leads to nan values in pyclesperanto and to 0 in clij2.

We need to discuss how to deal with nan values.

Cheers,
Robert

**Comment by jni** [comment]
pyclesperanto seems to have it right here, no? =)

**Comment by haesleinhuepf** [comment]
Yes, I'd say so. I'm just not sure why CLIJ does it as it does. Maybe, [because float[] arrays](https://github.com/clij/clij-core/blob/master/src/main/java/net/haesleinhuepf/clij/converters/implementations/ClearCLBufferToImagePlusConverter.java#L69-L79) don't support NaN.

**Comment by haesleinhuepf** [comment]
We may have to add a NaN-to-zero filter to enable rapid conversion. This is relevant, e.g. because napari crashes if you hand over a 3D image for visualisation which contains NaN pixels

**Comment by jni** [comment]
> We may have to add a NaN-to-zero filter to enable rapid conversion

NumPy has you covered:

https://numpy.org/doc/stable/reference/generated/numpy.nan_to_num.html

=D

> e.g. because napari crashes if you hand over a 3D image for visualisation which contains NaN pixels

Yes but there are already [plans to improve this](https://github.com/vispy/vispy/pull/1769) in VisPy. So I think we should leave the "correct" nans there and point users who need it to `nan_to_num`.

**Comment by haesleinhuepf** [comment]
> NumPy has you covered

Just in case someone wants to do this in a cle workflow without pushing/pulling ;-)
https://github.com/clij/clij2/commit/810bbc56ed6ec94c0e2c098f98c8687aaccc3ff3

**Comment by haesleinhuepf** [comment]
> Yes but there are already [plans to improve this](https://github.com/vispy/vispy/pull/1769) in VisPy. So I think we should leave the "correct" nans there and point users who need it to `nan_to_num`.

Good to know. I was wondering if there is a policy. After two intense python weeks I'm getting the concept: Catch the error where it comes from, not later. Makes sense. It's just unfamiliar. In ImageJ many functions check for many issues causing even more issues and side-effects.
